3d/scripts/generate_fonts.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+
+#   Copyright 2020 Scott Bezek and the splitflap contributors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import logging
+import os
+import subprocess
+import sys
+
+from svg_processor import SvgProcessor
+from projection_renderer import Renderer
+
+script_dir = os.path.dirname(os.path.abspath(__file__))
+source_parts_dir = os.path.dirname(script_dir)
+repo_root = os.path.dirname(source_parts_dir)
+sys.path.append(repo_root)
+
+from util import rev_info
+
+if __name__ == '__main__':
+    logging.basicConfig(level=logging.DEBUG)
+
+    parser = argparse.ArgumentParser(add_help=False)  # avoid conflict with '-h' for height
+    parser.add_argument('-f', '--font', type=str, help='Name of font to use (no spaces)')
+    parser.add_argument('-t', '--text', type=str, help='String of text to generate')
+
+    parser.add_argument('-h', '--height', type=float, help='Font height, as scalar of total flap height')
+    parser.add_argument('-w', '--width', type=float, help='Font width, as scalar of total flap width')
+    parser.add_argument('-x', '--offset-x', type=float, help='Character offset from center, X-axis')
+    parser.add_argument('-y', '--offset-y', type=float, help='Character offset from center, Y-axis')
+
+    parser.add_argument('--ncolumns', type=int, help='Number of columns / characters per row')
+    parser.add_argument('--no-comp', action='store_false', default=True, help='Don\'t compensate for the gap between top and bottom flaps')
```

Comment:
nit: it's slightly confusing that the flag is called "no comp" but defaults to True and stores False when specified, and is then used as `'letter_gap_comp' : args.no_comp,`. It functions correctly, but looks like a bug since "comp" is being set to the value of "no comp". Can you make it `store_true`, default to `False`, and use it as `letter_gap_comp: not args.no_comp`? Alternatively, I think you could set `dest` to "comp" so the value would be referenced as "args.comp" instead of "args.no_comp", but that still seems slightly confusing to me.

Reply:
Agreed, changed.